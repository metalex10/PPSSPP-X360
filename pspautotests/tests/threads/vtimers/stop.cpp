#include "shared.h"

extern "C" int main(int argc, char *argv[]) {
	SceUID vtimer = sceKernelCreateVTimer("test", NULL);

	checkpointNext("Objects");
	checkpoint("  Normal: %08x", sceKernelStopVTimer(vtimer));
	checkpoint("  Twice: %08x", sceKernelStopVTimer(vtimer));
	checkpoint("  NULL: %08x", sceKernelStopVTimer(0));
	checkpoint("  Invalid: %08x", sceKernelStopVTimer(0xDEADBEEF));
	sceKernelDeleteVTimer(vtimer);
	checkpoint("  Deleted: %08x", sceKernelStopVTimer(vtimer));

	vtimer = sceKernelCreateVTimer("test", NULL);
	checkpointNext("Resetting");

	sceKernelStartVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStopVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStartVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStopVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStartVTimer(vtimer);
	sceKernelDelayThread(1000);
	sceKernelStopVTimer(vtimer);

	checkpoint("  Start/Stop 3 times: %dms", (int)(sceKernelGetVTimerTimeWide(vtimer) / 1000));
	checkpoint("  Base: %lld", sceKernelGetVTimerBaseWide(vtimer));

	sceKernelDeleteVTimer(vtimer);
	return 0;
}