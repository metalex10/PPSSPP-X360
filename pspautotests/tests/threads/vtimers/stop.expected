[x] Objects
[x]   Normal: 00000000
[x]   Twice: 00000000
[x]   NULL: 800201bf
[x]   Invalid: 800201be
[x]   Deleted: 800201be

[x] Resetting
[r]   Start/Stop 3 times: 3ms
[x]   Base: 0
